src/backend/distributed/shared_library_init.c

```diff
@@ -2226,6 +2226,19 @@ CitusAuthHook(Port *port, int status)
 									  "regular client connections",
 									  MaxClientConnections)));
 		}
+
+		/*
+		 * Right after this, before we assign global pid, this backend
+		 * might get blocked by a DDL as that happens during parsing.
+		 *
+		 * That's why, lets mark the backend as an external backend
+		 * which is likely to execute a distributed command.
+		 *
+		 * We do this so that this backend gets the chance to show
+		 * up in citus_lock_waits.
+		 */
+		InitializeBackendData();
+		OverrideBackendDataDistributedCommandOriginator(true);
```

Comment:
Is Override still the right term / this the right name? iirc this was mainly for testing at first.

Reply:
I changed it to `SetBackendDataDistributedCommandOriginator`